service/awsconfig/v8/resource/lifecycle/current.go

```diff
@@ -95,34 +101,212 @@ func (r *Resource) GetCurrentState(ctx context.Context, obj interface{}) (interf
 		}
 	}
 
-	// TODO check if NodeConfig exists
-	// TODO create NodeConfig if it does not exist
-	// TODO check if NodeConfig has final state
+	{
+		for _, instance := range instances {
+			r.logger.LogCtx(ctx, "level", "debug", "message", "looking for node config for the guest cluster")
+
+			privateDNS, err := r.privateDNSForInstance(*instance.InstanceId)
+			if err != nil {
+				return nil, microerror.Mask(err)
+			}
+
+			n := v1.NamespaceDefault
+			o := metav1.GetOptions{}
+
+			_, err = r.g8sClient.CoreV1alpha1().NodeConfigs(n).Get(privateDNS, o)
+			if errors.IsNotFound(err) {
+				r.logger.LogCtx(ctx, "level", "debug", "message", "did not find node config for guest cluster node")
+
+				err := r.createNodeConfig(ctx, customObject, *instance.InstanceId, privateDNS)
+				if err != nil {
+					return nil, microerror.Mask(err)
+				}
+
+			} else if err != nil {
+				return nil, microerror.Mask(err)
+			} else {
+				r.logger.LogCtx(ctx, "level", "debug", "message", "found node config for the guest cluster")
 
-	for _, instance := range instances {
-		r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("completing lifecycle hook action for guest cluster instance '%s'", *instance.InstanceId))
+				r.logger.LogCtx(ctx, "level", "debug", "message", "waiting for inspection of node config for the guest cluster")
+			}
+		}
+	}
+
+	{
+		r.logger.LogCtx(ctx, "level", "debug", "message", "start inspection of node configs for the guest cluster")
 
-		i := &autoscaling.CompleteLifecycleActionInput{
-			AutoScalingGroupName:  aws.String(workerASGName),
-			InstanceId:            instance.InstanceId,
-			LifecycleActionResult: aws.String("CONTINUE"),
-			LifecycleHookName:     aws.String(key.NodeDrainerLifecycleHookName),
+		n := v1.NamespaceDefault
+		o := metav1.ListOptions{
+			LabelSelector: fmt.Sprintf("%s=%s", key.ClusterIDLabel, key.ClusterID(customObject)),
 		}
 
-		_, err := r.aws.AutoScaling.CompleteLifecycleAction(i)
+		nodeConfigs, err := r.g8sClient.CoreV1alpha1().NodeConfigs(n).List(o)
 		if err != nil {
 			return nil, microerror.Mask(err)
 		}
 
-		r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("completed lifecycle hook action for guest cluster instance '%s'", *instance.InstanceId))
+		if len(nodeConfigs.Items) > 0 {
+			for _, nodeConfig := range nodeConfigs.Items {
+				r.logger.LogCtx(ctx, "level", "debug", "message", "inspecting node config for the guest cluster")
+
+				if hasFinalStatus(nodeConfig.Status.Conditions) {
+					r.logger.LogCtx(ctx, "level", "debug", "message", "node config of guest cluster has final state")
+				} else {
+					r.logger.LogCtx(ctx, "level", "debug", "message", "node config of guest cluster has no final state")
+					continue
+				}
+
+				instanceID, err := instanceIDFromAnnotations(nodeConfig.GetAnnotations())
+				if err != nil {
+					return nil, microerror.Mask(err)
+				}
```

Comment:
This is a special thing for AWS. We use annotations to transport EC2 instance IDs. Otherwise the lookups of all necessary information again would be quite a ball ache.

Reply:
Done.